pageserver/src/tenant/secondary/downloader.rs

```diff
@@ -1079,12 +1117,12 @@ impl<'a> TenantDownloader<'a> {
                 }
             }
 
-            if on_disk.metadata.generation_file_size() != on_disk.metadata.generation_file_size() {
+            if on_disk.metadata.generation_file_size() != layer.metadata.generation_file_size() {
```

Comment:
This is an unrelated bug fix, right?  It looks like the old condition was comparing a number to itself

Reply:
yeah, I guess this was a mistake from #10529. I can make a separate PR if we don't end up merging this one.